### ISSUE TITLE
incorrect result with timestamp offset modifier

### ISSUE THREAD

**Issue by mxlxm** [VictoriaMetrics/VictoriaMetrics#415]
**Describe the bug**
Offset modifier been ignored

**To Reproduce**
```
timestamp(up{job="prometheus"} offset 1h)
time() - timestamp(up{job="prometheus"} offset 2d)
```

**Expected behavior**
correct result.

**Screenshots**

**Version**
```
$ ./vmstorage -version
vmstorage-20200213-163134-tags-v1.33.1-cluster-0-gafecb344
```

**Comment by valyala** [comment]
Confirmed the bug. Temporary workaround is to put `offset` outside `timestamp()` call:

```
timestamp(up{job="prometheus"}) offset 1h
time() - (timestamp(up{job="prometheus"}) offset 2d)
```

**Comment by valyala** [comment]
The bug should be fixed in the commit 6f7f64f757b27740d3d3644677a836c274fa9cc9 . This commit will be included in the next release.

@mxlxm , could you build VictoriaMetrics from 6f7f64f757b27740d3d3644677a836c274fa9cc9 commit according to [the docs](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/README.md#how-to-build-from-sources) and verify whether the issue is fixed there?

**Comment by mxlxm** [comment]
@valyala verified, it works. Great job!

**Comment by valyala** [comment]
The bugfix is available starting from [v1.34.8](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as fixed.
